src/pytorch_lightning/strategies/fully_sharded_native.py

```diff
@@ -358,7 +362,7 @@ def get_registered_strategies(cls) -> List[str]:
 
     @classmethod
     def register_strategies(cls, strategy_registry: Dict) -> None:
-        if _TORCH_GREATER_EQUAL_1_12:
+        if _TORCH_GREATER_EQUAL_1_12 and torch.distributed.is_available():
```

Comment:
Why add this? What's the stacktrace if you remove it?

Reply:
```
"lightning/src/pytorch_lightning/strategies/fully_sharded_native.py", line 379, in register_strategies
    cpu_offload=CPUOffload(offload_params=True),
TypeError: 'NoneType' object is not callable
```

`CPUOffload` is defined as None when torch>=1.12.0 and torch.distributed is not available.